bin/src/_main.rs

```diff
@@ -148,134 +150,110 @@ pub async fn _main(
     let mut k8s_claimed_lease: Option<String> = None;
     let mut metrics_stream_feature_meta: Option<FeatureLeaderMeta> = None;
 
-    let (k8s_event_stream, metric_stats_stream) = match create_k8s_client_default_from_env(
-        user_agent.clone(),
-    ) {
-        Ok(k8s_client) => {
-            info!("K8s Config Startup Option: {:?}", &config.startup);
-            check_startup_lease_status(
-                Some(&config.startup),
-                &mut k8s_claimed_lease,
-                k8s_client.clone(),
-            )
-            .await;
+    let (k8s_event_stream, metric_stats_stream) =
+        match create_k8s_client_default_from_env(user_agent.clone()) {
+            Ok(k8s_client) => {
+                info!("K8s Config Startup Option: {:?}", &config.startup);
+                check_startup_lease_status(
+                    Some(&config.startup),
+                    &mut k8s_claimed_lease,
+                    k8s_client.clone(),
+                )
+                .await;
+
+                if config.log.use_k8s_enrichment == K8sTrackingConf::Always
+                    && std::env::var_os("KUBERNETES_SERVICE_HOST").is_some()
+                {
+                    let node_name = std::env::var("NODE_NAME").ok();
+                    match K8sMetadata::new(k8s_client.clone(), node_name.as_deref()).await {
+                        Ok((driver, v)) => {
+                            tokio::spawn(driver);
+                            executor.register(v);
+                            info!("Registered k8s metadata middleware");
+                        }
+                        Err(e) => {
+                            let message = format!(
+                                "The agent could not access k8s api after several attempts: {}",
+                                e
+                            );
+                            error!("{}", message);
+                            panic!("{}", message);

```

Comment:
Why the need for the error and the panic?

Reply:
It was already there just looks new because of some moving code. 

178 in the diff, that being said.... I don't really know...